x/gamm/simulation/sim_msgs.go

```diff
@@ -44,9 +39,57 @@ func RandomJoinPoolMsg(k keeper.Keeper, sim *simulation.SimCtx, ctx sdk.Context)
 	}, nil
 }
 
-func RandomCreateUniv2PoolMsg(k keeper.Keeper, sim *simulation.SimCtx, ctx sdk.Context) (*balancertypes.MsgCreateBalancerPool, error) {
-	// 1) Select two denoms, ideally with some frequency weighting based on distribution amongst addrs
-	// 2) Select sender with both denoms + creation fee
-	// 3) Create pool
-	return nil, errors.ErrNotImplemented
+// RandomExitPoolMsg pseudo-randomly selects an existing pool ID, attempts to find an account with the
+// respective unbonded gamm shares, and attempts to execute an exit pool transaction
+func RandomExitPoolMsg(k keeper.Keeper, sim *simulation.SimCtx, ctx sdk.Context) (*gammtypes.MsgExitPool, error) {
+	// select a pseudo-random pool ID, max bound by the upcoming pool ID
+	pool_id := simulation.RandLTBound(sim, k.GetNextPoolNumber(ctx))
+	pool, err := k.GetPoolAndPoke(ctx, pool_id)
+	if err != nil {
+		return &gammtypes.MsgExitPool{}, err
+	}
+	// select an address that has gamm shares of the selected pool
+	gammDenom := fmt.Sprintf("gamm/pool/%v", pool_id)
+	sender, gammShares, senderExists := sim.SelAddrWithDenom(ctx, gammDenom)
+	if !senderExists {
+		return &gammtypes.MsgExitPool{}, fmt.Errorf("no sender with denom %s exists", gammDenom)
+	}
+	// calculate the minimum number of tokens received from input of gamm shares
+	tokenOutMins, _ := pool.CalcExitPoolCoinsFromShares(ctx, gammShares.Amount, pool.GetExitFee(ctx))
+	return &gammtypes.MsgExitPool{
+		Sender:        sender.Address.String(),
+		PoolId:        pool_id,
+		ShareInAmount: gammShares.Amount,
+		TokenOutMins:  tokenOutMins,
+	}, nil
+}
+
+// RandomCreatePoolMsg attempts to find an account with two or more distinct denoms and attempts to send a
+// create pool message composed of those denoms
+func RandomCreatePoolMsg(k keeper.Keeper, sim *simulation.SimCtx, ctx sdk.Context) (*balancertypes.MsgCreateBalancerPool, error) {
```

Comment:
```suggestion
func RandomCreateUniV2lMsg(k keeper.Keeper, sim *simulation.SimCtx, ctx sdk.Context) (*balancertypes.MsgCreateBalancerPool, error) {
```

UniV2 = Balancer with two denoms, and 50-50 weight!